### PR TITLE
fix(rbac): watch users and permission-policies

### PR DESCRIPTION
**Pull request by debsmita1**
Resolves:
https://github.com/janus-idp/backstage-plugins/issues/1101

GIF:

https://github.com/janus-idp/backstage-plugins/assets/22490998/388b7c2d-5e64-498c-9e61-06ce536a5c6e

### CODE REVIEW

**Comment by ciiay** [comment]
Hi @debsmita1 This is probably not in this scope, I have tested the scenario where I edited permission policies on the Edit Role page, the role Overview page picks up the change but the RBAC role list page didn't get updated. Looks like the `RolesData.permissions` value is not being watched on the role list page. Let me know if I'm missing anything here.


https://github.com/janus-idp/backstage-plugins/assets/26255262/7cb10a7b-aa7d-463d-800b-9351f3c977b4

File: plugins/rbac/src/types.ts [hunk 1]
```diff
@@ -17,7 +17,7 @@ export type RolesData = {
 
 export type MembersData = {
   name: string;
-  type: 'User' | 'Group';
+  type: string;
```
Comment (ciiay) [inline plugins/rbac/src/types.ts:20]: Just out of curiosity why this change? Is there a value other than `'User' | 'Group'` ?

File: plugins/rbac/src/hooks/usePermissionPolicies.ts [hunk 2]
```diff
@@ -1,50 +1,80 @@
...
-        name: (permissionPolicies as Response)?.status,
-        message: `Error fetching permission policies. ${(
-          permissionPolicies as Response
-        )?.statusText}`,
-      },
+    retry: { policiesRetry, permissionPoliciesRetry },
+    error:
+      (policiesError as Error) ||
+      (permissionPoliciesError as Error) ||
+      getErrorText(policies, permissionPolicies),
```
Comment (ciiay) [inline plugins/rbac/src/hooks/usePermissionPolicies.ts:78]: Do we need type assertion here? `policiesError` and `permissionPoliciesError` are already in Error type.

**Review by ciiay — CHANGES_REQUESTED** [review]
Tested working good. Just left a few small comments.

File: plugins/rbac/src/hooks/usePermissionPolicies.ts (hunk 2 above)
Comment (debsmita1) [reply plugins/rbac/src/hooks/usePermissionPolicies.ts:78]: good catch. Fixed it.

**Review by invincibleJai — APPROVED** [review]
/lgtm
